packages/block-editor/src/components/block-caption/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import RichText from '../rich-text/';
+
+// eslint-disable-next-line jsdoc/require-param
+/**
+ * Block caption component which is a thin wrapper around the RichText component
+ * that can be used to render captions for blocks like image block, video block,
+ * table block, embed block, gallery block e.t.c
+ *
+ * @param {Object} props Component props.
+ * @return {WPElement} Block Caption.
+ */
+
+const BlockCaption = ( {
```

Comment:
Added a `BlockCaption` component to avoid duplicate and redundant code in all blocks that have a caption